### DISCUSSION TITLE
printf on "other" serial ports

### DISCUSSION THREAD

**Discussion by microPaul** [SpenceKonde/DxCore#300] [General]
I may be in the minority, but for some situations I like to use printf() statements in my code for AVR-DB devices.  My problem is that I don't know what syntax to use (or even if it's possible) to cause printf() statements to print on UART2 or UART5.  I know that I can sort of get the effect with sprintf(p, "blah", blah) and then follow that with Serial5.print(p).  But I wonder if there's a simpler way to cause printf() to print on UART 5.

I appreciate all guidance that may be offered.

Paul

**Comment by SpenceKonde** [comment]
Are you saying that Serial.printf() works , but Serial2.printf() or Serial5.printf() doesn't behave identically? 

If so please open an issue with code that reproduces the issue. printf should work on all of the serial ports in exactly the same way

**Comment by microPaul** [comment]
Ah.... this is somewhat embarrassing.  Upon further testing it appears that printf does work correctly on Dx Core when one uses the correct syntax - SerialX.printf("blah", blah) .  When the wrong syntax is used then it doesn't work so well.  Thanks for the correction.
